### ISSUE TITLE
Context configuration by directory

### ISSUE THREAD

**Issue by Anonymous-Lurker** [aptakube/aptakube#259]
### What problem are you trying to solve? Do you a suggestion on how to solve it?

Could we have a feature to manage contexts by directory?  I have more than 500 context files that I deal with.  I have automation that generates those files (this automation is shared with a larger team, many who don’t use Kubernetes on a regular basis at all).  Customizing the files to add aptakube extension information would be more time consuming for me than placing the files into a hierarchical directory structure on disk.  My Kube context files follow a naming convention and so it would be very easy for me to move those files around and I would not have to worry about inserting aptakube specific information into the context files themselves.  Maybe either a filter view that lets me select configs by directory and then by regex?  A hierarchical view of the contexts as they exist on disk would be awesome as well.  I’d be happy to contribute to a bug bounty.

**Comment by goenning** [comment]
that's a lot of contexts :)

My initial reaction was to suggest using Context Groups via context extension, especially because you already have some automation to generate those files. I don't want to add something that's too specific to a single user, but how does the hierarchical directory structure look like? 

That'd also require a change to how we load context files, as we currently don't load files on sub directories.

--- 

btw, this filter supports fuzzysearch. Does that help?

![image](https://github.com/user-attachments/assets/1c7c3d0d-b6f4-408d-aad9-94d2e7fb5c00)

**Comment by Anonymous-Lurker** [comment]
Yeah the fuzzy search helps for sure, maybe there could be presets?  Would it also be possible to show the contexts in a way that allows for the adjustment of text size?  For instance even with the fuzzy search matching some substrings I'll still end up with quite a large list.   I can look at adding the aptakube extension information into the context files, I was just hoping that an on disk layout would be possible.  But I can play with the aptakube extension stuff.  Any  additions to the fuzzy search and controlling the text sizes in the context picker area would be nice.  Is the number of contexts that I work with fairly unusual?   I'm supporting a bunch of different clusters for different customer properties.   Another interesting feature might be a toggle in the UI to make any operation that is a modify or write something that can be disabled globally.  I'd like to have the UI open in like a read only mode most of the time, if that makes sense.  If I add read-only permissions that will add more contexts.

**Comment by Anonymous-Lurker** [comment]
Maybe fuzzy search "favorites" so that favorites can be saved and clicked on/modified without modifying the config files themselves?

**Comment by Anonymous-Lurker** [comment]
Sorry, I forgot to answer the directory question.  So it would be like ~/.kube/customer_name/permissions/property_subtype (some of the subtypes are related to internal processing, some of the clusters are specialized and separated for data handling concerns, others are serving things to the internet).
